docs/src/main/asciidoc/mailer-reference.adoc

```diff
@@ -0,0 +1,351 @@
+
+There are 2 APIs:
+
+* `io.quarkus.mailer.Mailer` provides the imperative (blocking and synchronous) API;
+* `io.quarkus.mailer.reactive.ReactiveMailer` provides the reactive (non-blocking and asynchronous) API
+
+NOTE: The two APIs are equivalent feature-wise. Actually the `Mailer` implementation is built on top of the `ReactiveMailer` implementation.
+
+[NOTE]
+.Deprecation
+====
+`io.quarkus.mailer.ReactiveMailer` is deprecated in favor of `io.quarkus.mailer.reactive.ReactiveMailer`.
+====
+
+To send a simple email, proceed as follows:
+
+[source, java]
+----
+// Imperative API:
+mailer.send(Mail.withText("to@acme.org", "A simple email from quarkus", "This is my body."));
+// Reactive API:
+Uni<Void> stage = reactiveMailer.send(Mail.withText("to@acme.org", "A reactive email from quarkus", "This is my body."));
+----
+
+For example, you can use the `Mailer` in an HTTP endpoint as follows:
+
+[source, java]
+----
+@GET
+@Path("/imperative")
+public void sendASimpleEmail() {
+    mailer.send(Mail.withText("to@acme.org", "A simple email from quarkus", "This is my body"));
+}
+
+@GET
+@Path("/reactive")
+public Uni<Void> sendASimpleEmailAsync() {
+    return reactiveMailer.send(
+            Mail.withText("to@acme.org", "A reactive email from quarkus", "This is my body"));
+}
+----
+
+== Creating Mail objects
+
+The mailer lets you send `io.quarkus.mailer.Mail` object.
+You can create new `io.quarkus.mailer.Mail` instances from the constructor or from the `Mail.withText` and
+`Mail.withHtml` helper methods.
+The `Mail` instance lets you add recipients (to, cc, or bcc), set the subject, headers, sender (from) address...
+
+You can also send several `Mail` objects in one call:
+
+[source, java]
+----
+mailer.send(mail1, mail2, mail3);
+----
+
+[[attachments]]
+== Sending attachments
+
+To send attachments, just use the `addAttachment` methods on the `io.quarkus.mailer.Mail` instance:
+
+[source,java]
+----
+@GET
+@Path("/attachment")
+public void sendEmailWithAttachment() {
+        mailer.send(Mail.withText("clement.escoffier@gmail.com", "An email from quarkus with attachment",
+                "This is my body")
+                .addAttachment("my-file-1.txt",
+                        "content of my file".getBytes(), "text/plain")
+                .addAttachment("my-file-2.txt",
+                        new File("my-file.txt"), "text/plain")
+        );
+}
+----
+
+Attachments can be created from raw bytes (as shown in the snippet) or files.
+Note that files are resolved from the working directory of the application.
+
+[[html]]
+== Sending HTML emails with inlined attachments
+
+When sending HTML emails, you can add inlined attachments.
+For example, you can send an image with your email, and this image will be displayed in the mail content.
+If you put the image file into resources folder, you should specify the full path to the file, _e.g._ `META-INF/resources/quarkus-logo.png` otherwise Quarkus will look for the file in the root directory.
```

Comment:
```suggestion
If you put the image file into the `META-INF/resources` folder, you should specify the full path to the file, _e.g._ `META-INF/resources/quarkus-logo.png` otherwise Quarkus will look for the file in the root directory.
```